OpenEdXMobile/src/main/java/org/edx/mobile/base/MainApplication.java

```diff
@@ -133,24 +133,29 @@ private void init() {
     private void checkIfAppVersionUpgraded(Context context) {
         PrefManager.AppInfoPrefManager prefManager = new PrefManager.AppInfoPrefManager(context);
         long previousVersionCode = prefManager.getAppVersionCode();
-        final long currVersionCode = (long) BuildConfig.VERSION_CODE;
-        if (previousVersionCode < currVersionCode) {
+        final long curVersionCode = (long) BuildConfig.VERSION_CODE;
+        if (previousVersionCode < 0) {
```

Comment:
1. `repairDownloadCompletionData`don't need to be called on app's fresh installation time as it mentioned in the documents in its implementation.
2. We should avoid clearing of app related shared preferences (as they don't belong to user) on logging out.
3. This case is very very less likely to happen, we can ignore it.

Only option 2 is considerable. I am figuring out its best solution currently.